hypothesis-python/src/hypothesis/database.py

```diff
@@ -675,6 +677,43 @@ def delete(self, key: bytes, value: bytes) -> None:
         raise RuntimeError(self._read_only_message)
 
 
+class BackgroundWriteDatabase(ExampleDatabase):
+    """A wrapper which defers writes on the given database to a background thread.
+    Reads are still executed synchronously on the calling thread.
+    """
+
+    def __init__(self, db: ExampleDatabase) -> None:
+        self._db = db
+        self._queue: Queue[tuple[str, tuple[bytes, ...]]] = Queue()
+        self._thread = Thread(target=self._worker, daemon=True)
+        self._thread.start()
+
+    def __repr__(self) -> str:
+        return f"BackgroundWriteDatabase({self._db!r})"
+
+    def _worker(self) -> None:
+        while True:
+            method, args = self._queue.get()
+            getattr(self._db, method)(*args)
+            self._queue.task_done()
+
+    def fetch(self, key: bytes) -> Iterable[bytes]:
+        return self._db.fetch(key)
+
+    def save(self, key: bytes, value: bytes) -> None:
+        self._queue.put(("save", (key, value)))
+
+    def delete(self, key: bytes, value: bytes) -> None:
+        self._queue.put(("delete", (key, value)))
+
+    def move(self, src: bytes, dest: bytes, value: bytes) -> None:
+        self._queue.put(("move", (src, dest, value)))
+
+    def join(self) -> None:
+        """Waits until all queued writes on the background thread have been processed."""
+        self._queue.join()
```

Comment:
Weak preference to avoid making this a public method; `def _join` or just calling `db._queue.join()` might be good alternatives.